src/plugins/data_source/server/saved_objects/data_connection.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObjectsType } from 'opensearch-dashboards/server';
+
+export const dataConnection: SavedObjectsType = {
+  name: 'data-connection',
+  namespaceType: 'agnostic',
+  hidden: false,
+  management: {
+    icon: 'apps', // todo: pending ux #2034
+    defaultSearchField: 'connectionId',
+    importableAndExportable: true,
+    getTitle(obj) {
+      return obj.attributes.title;
+    },
+    getEditUrl(obj) {
+      return `/management/opensearch-dashboards/dataConnections/${encodeURIComponent(obj.id)}`;
+    },
+    getInAppUrl(obj) {
+      return {
+        path: `/app/management/opensearch-dashboards/dataConnections/${encodeURIComponent(obj.id)}`,
+        uiCapabilitiesPath: 'management.opensearchDashboards.dataConnections',
```

Comment:
Guess you need to add the capability `management.opensearchDashboards.dataConnections` to saved_object_managements plugin?